documentation-website/Writerside/topics/Deep-Dive-into-DSL.md

```diff
@@ -766,3 +766,65 @@ The values specified in the statement block will be used for the insert statemen
 In the example above, if the original row was inserted with a user-defined <code>rating</code>, then <code>replace()</code> was executed with a block that omitted the <code>rating</code> column, 
 the newly inserted row would store the default rating value. This is because the old row was completely deleted first.
 </note>
+
+## Column transformation
+
+Column transformations allow to define custom transformations between database column types and application's data types. 
+This can be particularly useful when you need to store data in one format but work with it in another format within your application.
+
+Consider the following example, where we define a table to store meal times and transform these times into meal types:
+
+```kotlin
+enum class Meal {
+    BREAKFAST,
+    LUNCH,
+    DINNER
+}
+
+object Meals : Table() {
+    val mealTime: Column<Meal> = time("meal_time")
+        .transform(
+            toTarget = {
```

Comment:
Let's discuss the naming. Please define the `Target` upfront

Reply:
If you want to have a symmetric name for `toColumn`, I would go with `toUserType`. 

There are also other options we can discuss:
- pack / unpack
- save / load
- encode / decode
- wrap / unwrap
- persist / retrieve
- store / fetch